src/fireEvent.js

```diff
@@ -33,12 +33,17 @@ const isPointerEventEnabled = (
   return isPointerEventEnabled(element.parent, true);
 };
 
+const isTouchEvent = (eventName?: string) => {
```

Comment:
The name `isTouchEvent` feels odd given that `onTouch*` is not considered to be `isTouchEvent` here. However, I chose the name `isTouchEvent` based upon the language in the [`pointerEvents` documentation](https://reactnative.dev/docs/view#pointerevents). Open to feedback on this naming.

Reply:
I'm not sure whether this list is complete and/or even can be a static one because it may be platform-dependent.

On Android setting `pointerEvents` works by intercepting and discarding `MotionEvent`s. These include  at least `touchstart`, `touchend`, `touchmove` and `touchcancel` events. See: https://github.com/facebook/react-native/blob/main/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java#L220-L245.

In case of iOS it is setting this: https://developer.apple.com/documentation/uikit/uiview/1622577-userinteractionenabled?language=objc (related RN code: https://github.com/facebook/react-native/blob/main/React/Views/RCTView.m#L151-L158) to `false`, so it includes `touch, press, keyboard, and focus events`.

I'd experiment with both platforms and see what superset of events is covering both platforms.

Reply:
That seems like a good approach for me. @thymikee has a final say here though!

Reply:
Still haven't found time for this, please bare with me! :D

Reply:
Hey @thymikee. 👋🏻 I wanted to check back in with you. I am still interested merging this work to fix some tests in [WordPress/Gutenberg](https://github.com/wordpress/gutenberg). It is completely understandable if you do not have time to review. I merely wanted to make sure the PR was not forgotten. Thanks! 🙇🏻

Reply:
Hey @dcalhoun, I'll need more time to process it. Right now I focus on releasing a new library and needed to postpone this. It's not forgotten though.

In the meantime, I'll ask @AugustinLF, maybe he'll be able to check it.

Reply:
Thanks, @thymikee. That makes sense. I appreciate your time. 

P.S. - feel free to ignore my ping from a different PR.

Reply:
So if I understand correctly, your suggestion of implementation is strictly more correct than what we have, but is missing some additional cases? i.e. it makes things better, but doesn't fix everything, right?

If that's the case, I'm fine with merging this change. I'd like to have documentation (code and/or through a detailed issue) on the missing behaviour, so we can fix it going forward. Does that seem reasonable?

Reply:
I'd suggest to merge.